security/src/main/java/io/micronaut/security/rules/SensitiveEndpointRule.java

```diff
@@ -57,35 +76,122 @@
      *
      * @param endpointSensitivityProcessor The endpoint configurations
      */
-    SensitiveEndpointRule(EndpointSensitivityProcessor endpointSensitivityProcessor) {
+    public SensitiveEndpointRule(EndpointSensitivityProcessor endpointSensitivityProcessor) {
         this.endpointMethods = endpointSensitivityProcessor.getEndpointMethods();
     }
 
     @Override
     public SecurityRuleResult check(HttpRequest<?> request, @Nullable RouteMatch<?> routeMatch, @Nullable Map<String, Object> claims) {
         if (routeMatch instanceof MethodBasedRouteMatch) {
-            ExecutableMethod method = ((MethodBasedRouteMatch) routeMatch).getExecutableMethod();
-
+            ExecutableMethod<?, ?> method = ((MethodBasedRouteMatch<?, ?>) routeMatch).getExecutableMethod();
             if (endpointMethods.containsKey(method)) {
-                Boolean sensitive = endpointMethods.get(method);
-                if (claims == null && sensitive) {
-                    if (LOG.isDebugEnabled()) {
-                        LOG.debug("The endpoint method [{}] is sensitive and no authentication was found. Rejecting the request", method);
-                    }
-                    return SecurityRuleResult.REJECTED;
-                } else {
-                    if (LOG.isDebugEnabled()) {
-                        LOG.debug("The endpoint method [{}] is not sensitive or the request was authenticated. Allowing the request", method);
-                    }
-                    return SecurityRuleResult.ALLOWED;
-                }
+                return check(request, claims, method);
             }
         }
         return SecurityRuleResult.UNKNOWN;
     }
 
+    /**
+     * Evaluate the Endpoint's method.
+     * @param request HTTP Request
+     * @param claims Claims of authenticated user. null if the user is not authenticated
+     * @param method Route method
+     * @return The Result
+     */
+    @NonNull
+    protected SecurityRuleResult check(@NonNull HttpRequest<?> request,
+                                       @Nullable Map<String, Object> claims,
+                                       @NonNull ExecutableMethod<?, ?> method) {
+        Boolean sensitive = endpointMethods.get(method);
+        if (sensitive) {
+            if (claims == null) {
+                return checkSensitiveAnonymous(request, method);
+            }
+            return checkSensitiveAuthenticated(request, claims, method);
+        }
+        return checkNotSensitive(request, claims, method);
+    }
```

Comment:
I'm not sure if this is the best way to encapsulate everything calling the threes protected methods from here. All three methods are the same in the sense that they log a message and return the appropriate `SecurityRuleResult` but they don't check any condition. Everything is decided in this method.
Did you create those three methods in other for users to be able to override them without override this one?

Reply:
Yes, three protected methods makes easier the replacement. See example, in the tests which I show in the docs.

Reply:
:+1: